pkg/gin-middleware/oapi_validate.go

```diff
@@ -122,12 +122,12 @@ func ValidateRequestFromContext(c *gin.Context, router routers.Router, options *
 
 	// Pass the gin context into the request validator, so that any callbacks
 	// which it invokes make it available.
-	requestContext := context.WithValue(context.Background(), GinContextKey, c)
+	requestContext := context.WithValue(context.Background(), GinContextKey, c) //nolint:staticcheck
```

Comment:
SA1029: should not use built-in type string as key for value; define your own type to avoid collisions (staticcheck)